expenses-manager-backend/src/main/java/com/endava/expensesmanager/service/impl/CurrencyConversionServiceImpl.java

```diff
@@ -43,13 +35,35 @@ public BigDecimal convert(BigDecimal amount, String fromCurrency, String toCurre
             return amount;
         }
 
-        Map<String, Double> rates = conversionRates.get(fromCurrency);
-        if (rates == null || !rates.containsKey(toCurrency)) {
-            throw new BadRequestException("Unsupported currency conversion from " + fromCurrency + " to " + toCurrency);
+        String url = String.format("%s/pair/%s/%s", apiCurrencyUrl.replace("YOUR-API-KEY", apiKey), fromCurrency, toCurrency);
```

Comment:
aici faci replace la YOUR-API-KEY, dar in application.properties nu il ai in api url